src/main/java/org/broadinstitute/hellbender/tools/exome/cnlohcaller/CNLOHCaller.java

```diff
@@ -0,0 +1,752 @@
+                                             final double mafCredibleLow, final double mafCredibleHigh,
+                                             final double crCredibleMode, final double crCredibleLow,
+                                             final double crCredibleHigh, final double lambda) {
+
+
+        final int[] mVals = pis.keySet().stream().mapToInt(Integer :: intValue).toArray();
+        final int[] nVals = pis.keySet().stream().mapToInt(Integer :: intValue).toArray();
+
+        final double[][][] result = new double[rhos.length][mVals.length][nVals.length];
+        double total = 0.0;
+        for (int rhoi = 0; rhoi < rhos.length; rhoi++) {
+            for (int mi = 0; mi < mVals.length; mi++) {
+                for (int ni = 0; ni < nVals.length; ni++) {
+
+                    final double rho = rhos[rhoi];
+                    final int m = mVals[mi];
+                    final int n = nVals[ni];
+
+                    if ((rho < 0) || (rho > 1)) {
+                        result[rhoi][mi][ni] = 0;
+                    } else if ((rho == 0) && ((m != 1) || (n != 1))) {
+                        result[rhoi][mi][ni] = 0;
+                    } else if ((rho > 0) && (rho < rhoThreshold)) {
+                        result[rhoi][mi][ni] = 0;
+                    } else if ((rho != 0) && ((m == 1) && (n == 1))) {
+                        result[rhoi][mi][ni] = 0;
+                    } else {
+                        result[rhoi][mi][ni] = Math.exp(E_ln_phi_k[rhoi]) *
+                                pis.get(m) * pis.get(n) *
+                                calculateFmaf(rho, m, n, mafCredibleMode, mafCredibleLow, mafCredibleHigh, normalNumCopies) *
+                                calculateFcr(rho, m, n, lambda, crCredibleMode, crCredibleLow, crCredibleHigh,
+                                        segmentMeanVarianceInCR, normalNumCopies);
+                    }
+
+                    total += result[rhoi][mi][ni];
+                }
+            }
+        }
+
+        // Normalize by total
+        for (int rhoi = 0; rhoi < rhos.length; rhoi++) {
+            for (int mi = 0; mi < mVals.length; mi++) {
+                for (int ni = 0; ni < nVals.length; ni++) {
+                    result[rhoi][mi][ni] /= total;
+                }
+            }
+        }
+
+        return result;
+    }
+
+    protected double[] calc_E_ln_phi_k(final double E_alpha, final double[] eZskSummedOverSegs) {
+
+        final double eZskSummedOverAll = MathUtils.sum(eZskSummedOverSegs);
+
+        final double[] result = new double[eZskSummedOverSegs.length];
+        final int k = eZskSummedOverSegs.length;
+
+        // Initialize all pseudocounts to 1, except for index 0, which is 20;
+        //  This artificially increases the odds for a rho = 0.
+        final RealVector pseudocounts = new ArrayRealVector(eZskSummedOverSegs.length);
+        pseudocounts.set(1.0);
```

Comment:
Consider making the non-neutral pseudocount much smaller, possibly zero.  I'm worried that too large a value will counteract the sparsity of the Dirichlet.